cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1226,16 +1226,31 @@ CacheAllocator<CacheTrait>::findEviction(PoolId pid, ClassId cid) {
     ++searchTries;
 
     Item* candidate = itr.get();
+
+    // make sure no other thead is evicting the item
+    if (candidate->getRefCount() != 0 || candidate->isMoving()) {
+      ++itr;
+      continue;
+    }
+
+    if (candidate->isChainedItem()) {
+      candidate = &candidate->asChainedItem().getParentItem(compressor_);
+    }
+
+    auto toReleaseHandle = accessContainer_->find(*candidate);
+    if (!toReleaseHandle || !toReleaseHandle.isReady()) {
```

Comment:
We don't need to check `isReady()` since this we still hold LRU lock and this item cannot have been evicted to NvmCache.